src/main/java/elucent/albedo/lighting/LightManager.java

```diff
@@ -74,16 +76,38 @@ public static void update(World world) {
 			if (e instanceof ILightProvider){
 				addLight(((ILightProvider)e).provideLight());
 			}
+			if (e.hasCapability(Albedo.LIGHT_CAP, null)){
```

Comment:
It's not noticeably slower when used in the usual use-case of communication between blocks. But here, we're iterating over thousands of entities every *frame* - even something as simple as a HashMap lookup can add up quickly.

The problem is that capabilities cannot be cached. Ever. It's not in their API contract. You have to call hasCapability and getCapability *every single time*, whereas a tile entity generally will not magically remove or change their interface after it has been placed, allowing for caching and speedups to be added in the future.

You could design the light provider capability with an isInvalid() method so that it does end up being cacheable, though. But that's a bit more involved.